ast/src/impls.rs

```diff
@@ -57,7 +57,7 @@ impl<R> Expr<R> {
 #[cfg(target_arch = "x86_64")]
 static_assertions::assert_eq_size!(crate::Expr, [u8; 72]);
 #[cfg(target_arch = "x86_64")]
-static_assertions::assert_eq_size!(crate::Stmt, [u8; 136]);
+static_assertions::assert_eq_size!(crate::Stmt, [u8; 160]);
```

Comment:
I presume it is okay to bump this to the new size but am not sure what the implications of this assertion is.

Reply:
cc @MichaReiser

Reply:
The main reason for the assertion is to be aware of how changing the AST changes its size. This is important because ruff and RustPython hold many instances of these types and increasing their size mean:

* Each element in a `Vec` now requires more storage -> Overall increased memory consumption
* Reading or writing an element becomes slower because your computer must read or write more bytes. This is especially relevant for reading where the L1 cache short circuits reads by caching about 64 bytes of neighboring memory to avoid subsequent reads from (the very slow) memory. 

There's not much we can do about this here. We could consider changing this specific new field from `Option<TypeParam>` to `Option<Box<TypeParam>>` to reduce the size increase to 8 bytes, considering that only very few functions have type params. But I don't know if this is easily achievable with the asdl codegen.

Reply:
Since using `Box` is decided by `asdl_rs.py`, I think we can change the rule or manually specify a few types to be boxed. Do you have idea?

@zanieb It doesn't seem like coverage of this PR. Don't worry about it.

Reply:
I guess boxing it instead of `tinyvec` (which can cause problem with pyo3) could work as well.

Reply:
Tracking in https://github.com/RustPython/Parser/issues/94